src/cursor.c

```diff
@@ -891,15 +927,24 @@ cursor_axis(struct wl_listener *listener, void *data)
 	 */
 	struct seat *seat = wl_container_of(listener, seat, cursor_axis);
 	struct wlr_pointer_axis_event *event = data;
+	struct server *server = seat->server;
+	struct cursor_context ctx = get_cursor_context(server);
 	wlr_idle_notify_activity(seat->wlr_idle, seat->seat);
 
-	/* Make sure we are sending the events to the surface under the cursor */
-	cursor_update_focus(seat->server);
+	/* Bindings swallow mouse events if activated */
+	bool triggered_axis_binding = handle_cursor_axis(server, &ctx, event);
 
-	/* Notify the client with pointer focus of the axis event. */
-	wlr_seat_pointer_notify_axis(seat->seat, event->time_msec,
-		event->orientation, event->delta, event->delta_discrete,
-		event->source);
+	if (ctx.surface && !triggered_axis_binding) {
+		/* Notify client of axis event */
+
+		/* Make sure we are sending the events to the surface under the cursor */
+		cursor_update_common(server, &ctx, 0, false);
```

Comment:
Do we need `cursor_update_common()` here? 

The function `cursor_button_press()` does not have it at the equivalent place.
The `cursor_update_*()` functions took a long time to get right, so I'm a bit sensitive to adding additional calls unless we're very clear why.

Reply:
We need that to ensure that the mouse focus shifts to the the window under the cursor. It isn't a change in this PR, it was already in there before in form of `cursor_update_focus()`.

Reply:
@Consolatis thanks. Sorry I didn't think through properly.
It should be `cursor_update_common(server, &ctx, event->time_msec, false);`

Reply:
Post below updated